heron/cli/src/python/argparser.py

```diff
@@ -0,0 +1,143 @@
+class SubcommandHelpFormatter(argparse.RawDescriptionHelpFormatter):
+  def _format_action(self, action):
+    parts = super(argparse.RawDescriptionHelpFormatter, self)._format_action(action)
+    if action.nargs == argparse.PARSER:
+      parts = "\n".join(parts.split("\n")[1:])
+    return parts
+
+class HeronRCArgumentParser(argparse.ArgumentParser):
+	cmdmap=collections.defaultdict(dict)
+ 	def __init__(self, *args, **kwargs):
+		HeronRCArgumentParser.initializeFromRC()
+		super(HeronRCArgumentParser, self).__init__(*args, **kwargs)
+	
+	@classmethod
+	def remove_comments(hrc,string):
+		pattern = r"(\#.*$)"
+		# first group captures quoted strings (double or single)
+		# second group captures comments (//single-line or /* multi-line */)
+		regex = re.compile(pattern, re.MULTILINE|re.DOTALL)
+		def _replacer(match):
+		    # if the 2nd group (capturing comments) is not None,
+		    # it means we have captured a non-quoted (real) comment string.
+		    if match.group(1) is not None:
+		        return "" # so we will return empty to remove the comment
+		    else: # otherwise, we will return the 1st group
+		        return match.group(1) # captured quoted-string
+		return regex.sub(_replacer, string)
+
+	# initialize the command map from heron rc file in the parser, that can be later used for command substitution during parse_args phase
+	# patterns
+	# cmdmap [cmd][environment] = 'argument list'
+	# cmdmap ['submit']['devcluster/ads/PROD'] = '--config-property "heron.class.launcher=com.twitter.heron.scheduler.aurora.AuroraLauncher"'
+	# cmdmap ['*']['devcluster/ads/PROD'] = '--config-property "heron.class.launcher=com.twitter.heron.scheduler.aurora.AuroraLauncher"'
+	# cmdmap ['*']['*'] = '--config-property "heron.class.launcher=com.twitter.heron.scheduler.aurora.AuroraLauncher"'
+	@classmethod
+	def initializeFromRC(hrc):
+		if  len(hrc.cmdmap) > 0:
+			return 
+		if os.path.exists(HERON_RC):
+
+			with open(HERON_RC) as f:
+				hrc.cmdmap['*']['*'] = ''
+				for line in f:
+					m = p.match(line)
+					value = ''
+					command=''
+					env = ''
+					if (m != None):
+						value = hrc.remove_comments(m.group(3).rstrip(os.linesep))
+						command = ( m.group(1),'') [ m.group(1) == None or m.group(1) == '' ]
+						env = ( m.group(2), '') [ m.group(2) == None or m.group(2) == '']
+					else:
+						value = hrc.remove_comments(line.rstrip(os.linesep))
+
+					args_list = args.insert_bool_values(value.split())  # make sure that all the single args have a boolean value associated so that we can load the args to a key value structure
+					args_list_string = ' '.join (args_list)
+					if command == None or command == '' :
+						continue
+					if ( command in hrc.cmdmap and env in hrc.cmdmap[command]):
+						hrc.cmdmap[command][env] =  hrc.cmdmap[command][env] 	+ ' ' + args_list_string	
+					else:
+						hrc.cmdmap[command][env] =  args_list_string
+			#print "cmdmap", hrc.cmdmap	
+		else:
+			print "WARN: %s is not an existing file" % HERON_RC
```

Comment:
please use Log module under `heron/common` by adding `from heron.common.src.python.color import Log`, then you can use `Log.warn(...)`